src/app/modules/form/admin-form/admin-form.service.ts

```diff
@@ -1148,3 +1169,241 @@ const isMobileFieldUpdateAllowed = (
     },
   )
 }
+
+export const createTwilioCredentials = (
+  twilioCredentials: TwilioCredentials,
+  formId: string,
+): ResultAsync<unknown, ApplicationError> => {
+  const twilioCredentialsData: TwilioCredentialsData =
+    new TwilioCredentialsData(twilioCredentials)
+  const msgSrvcName = `formsg/${config.secretEnv}/form/${formId}/twilio`
+  const body: CreateSecretRequest = {
+    Name: msgSrvcName,
+    SecretString: twilioCredentialsData.toString(),
+  }
+
+  logger.info({
+    message: `no msgSrvcName, creating Twilio credentials for form ${formId}`,
+    meta: {
+      action: 'createTwilioCredentials',
+      formId: formId,
+      msgSrvcName,
+    },
+  })
+
+  return ResultAsync.fromPromise(
+    FormModel.startSession().then((session: ClientSession) =>
+      session
+        .withTransaction(
+          async () =>
+            await createTwilioTransaction(formId, msgSrvcName, body, session),
+        )
+        .then(() => session.endSession()),
+    ),
+    (error) => {
+      logger.error({
+        message: 'Error encountered when creating Twilio Secret',
+        meta: {
+          action: 'createTwilio',
```

Comment:
`logMeta` can avoid typos like this too, where the meta.action is not really the function's name